component-runtime-impl/src/main/java/org/talend/sdk/component/runtime/record/RecordConverters.java

```diff
@@ -142,6 +142,7 @@ private Record json2Record(final RecordBuilderFactory factory, final JsonObject
                         .withArray(factory
                                 .newEntryBuilder()
                                 .withName(key)
+                                .withLabel(key)
```

Comment:
“ withLabel should be automatically called by withName when sanitized name”“ : yes, i consider that already, but some risk, need more refactor, and so do like this, will keep this.

about your second question, already do

Reply:
> “ withLabel should be automatically called by withName when sanitized name”“ : yes, i consider that already, 
but some risk, need more refactor, and so do like this, will keep this.

I think doing like this is not good since it will call withLabel everytime and @undx doesn't want it.
For me the responsibility to call withLabel is the withName method when sanitized name != givenName.

If some connectors already call withName with a sanitized name, they will not have currently label. The will be fix after (if quick fix, maybe done within 1.1.15.1)